### PR TITLE
switch to generated password for jenkins service

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@spadgett @jwforres fyi.  I don't think this has a direct impact on the console except users won't know how to login anymore when you end them to the console....

@csrwng ptal

**Comment by jwforres** [comment]
On Tue, Aug 2, 2016 at 11:11 AM, Ben Parees <notifications@github.com>
wrote:

> @spadgett <https://github.com/spadgett> @jwforres
> <https://github.com/jwforres> fyi. I don't think this has a direct impact
> on the console except users won't know how to login anymore when you end
> them to the console....
>

And what are we going to do about that...

File: examples/jenkins/README.md [hunk 1]
```diff
@@ -68,8 +68,10 @@ Steps
 6. Create the sample application configuration
 
         $ oc new-app application-template.json
+
+    Note the generated password (JENKINS_PASSWORD) reported in the new-app output, you will need it to login to Jenkins.
```
Comment (csrwng) [inline examples/jenkins/README.md]: Should you include directions on how to get it if creating from the web console?

File: examples/jenkins/pipeline/README.md [hunk 2]
```diff
@@ -40,6 +40,14 @@ jenkins template represented by jenkinstemplate.json by running these commands a
     You should not need to access the jenkins console for anything, but if you want to configure settings or watch the execution,
     here are the steps to do so:
 
+    First you must determine the jenkins password which was randomly generated for you:
+
+        $ oc env dc/jenkins --list | grep JENKINS_PASSWORD
+
+    The value will appear as:
+
+        $ JENKINS_PASSWORD=faUXMOFUhRYXO2Gx
```
Comment (csrwng) [inline examples/jenkins/pipeline/README.md]: '$ ' is not part of the output

**Comment by csrwng** [comment]
Not related to this pull, but I instantiated Jenkins from the template and I wanted to delete it ... I've gotten used to just doing 'oc delete all -l app=blah' to delete anything, but I didn't see a common app label in the stuff created by the template

File: examples/jenkins/pipeline/README.md (hunk 2 above)
Comment (bparees) [inline examples/jenkins/pipeline/README.md]: good catch

File: examples/jenkins/README.md (hunk 1 above)
Comment (bparees) [inline examples/jenkins/README.md]: will update.

**Comment by bparees** [comment]
> And what are we going to do about that...

I have no idea :/

**Comment by bparees** [comment]
@csrwng it's got a common label of "template=jenkins-ephemeral-template", you don't see that?

however i'm going to change it to jenkins-pipelinetemplate.

File: examples/jenkins/pipeline/jenkinstemplate.json [hunk 3]
```diff
@@ -243,12 +244,13 @@
...
-      "description": "Password for the Jenkins user.",
+      "description": "Password for the Jenkins 'admin' user.",
       "generate": "expression",
-      "value": "password"
+      "from": "[a-zA-Z0-9]{16}",
+      "required": true
     }
   ],
   "labels": {
-    "template": "jenkins-ephemeral-template"
```
Comment (bparees) [inline examples/jenkins/pipeline/jenkinstemplate.json]: @csrwng

Comment (csrwng) [inline examples/jenkins/pipeline/jenkinstemplate.json]: ahh, yes, sorry I was describing the dc and looking at the wrong field. Works for me.

File: examples/jenkins/pipeline/samplepipeline.json [hunk 4]
```diff
@@ -82,11 +82,10 @@
       "kind": "Route",
       "apiVersion": "v1",
       "metadata": {
-        "name": "route-edge",
+        "name": "frontend",
         "creationTimestamp": null
       },
       "spec": {
-        "host": "www.example.com",
```
Comment (bparees) [inline examples/jenkins/pipeline/samplepipeline.json]: @csrwng

File: examples/jenkins/pipeline/samplepipeline.json [hunk 5]
```diff
@@ -82,11 +82,10 @@
       "kind": "Route",
       "apiVersion": "v1",
       "metadata": {
-        "name": "route-edge",
+        "name": "frontend",
```
Comment (bparees) [inline examples/jenkins/pipeline/samplepipeline.json]: @csrwng

**Comment by csrwng** [comment]
LGTM

**Comment by bparees** [comment]
[merge]

**Pull request merged**